packages/composite-checkout/src/components/billing-fields.js

```diff
@@ -0,0 +1,557 @@
+			/>
+			<DomainRegistrationLabel htmlFor="domain-registration">
+				Use your billing details for your domain registration contact information.
+			</DomainRegistrationLabel>
+		</DomainRegistrationCheckBoxWrapper>
+	);
+}
+
+DomainFieldsCheckbox.propTypes = {
+	toggleVisibility: PropTypes.func,
+};
+
+const DomainRegistrationCheckBoxWrapper = styled.div`
+	margin: 16px 0 24px;
+	display: flex;
+	width: 100%;
+`;
+
+const DomainRegistrationLabel = styled.label`
+	font-size: 14px;
+	color: ${props => props.theme.colors.textColor};
+	display: block;
+	position: relative;
+	padding-left: 5px;
+
+	:hover {
+		cursor: pointer;
+	}
+
+	:before {
+		display: block;
+		width: 16px;
+		height: 16px;
+		content: '';
+		position: absolute;
+		left: -20px;
+		top: 0;
+		border: 1px solid ${props => props.theme.colors.borderColorLight};
+		border-radius: 3px;
+	}
+`;
+
+const DomainRegistrationCheckbox = styled.input`
+	margin-right: 5px;
+	opacity: 0;
+
+	:checked + label:before {
+		background: ${props => props.theme.colors.highlight};
+	}
+
+	:checked + label:after {
+		display: block;
+		width: 4px;
+		height: 8px;
+		content: '';
+		position: absolute;
+		left: -14px;
+		top: 3px;
+		border-right: 2px solid ${props => props.theme.colors.surface};
+		border-bottom: 2px solid ${props => props.theme.colors.surface};
+		transform: rotate( 45deg );
+	}
+
+	:focus + label:before {
+		outline: ${props => props.theme.colors.outline} auto 5px;
+	}
+`;
+
+function AddressFields( { paymentData, setPaymentData, location } ) {
+	const localize = useLocalize();
+	const name = location + 'Name';
+	const address = location + 'Address';
+	const city = location + 'City';
+	const state = location + 'State';
+	const province = location + 'Province';
+
+	const {
+		billingName = '',
+		billingAddress = '',
+		billingCity = '',
+		billingState = '',
+		billingProvince = '',
+		billingNameError = null,
+		billingAddressError = null,
+		billingCityError = null,
+		billingStateError = null,
+		billingProvinceError = null,
+		domainsName = '',
+		domainsAddress = '',
+		domainsCity = '',
+		domainsState = '',
+		domainsProvince = '',
+		domainsNameError = null,
+		domainsAddressError = null,
+		domainsCityError = null,
+		domainsStateError = null,
+		domainsProvinceError = null,
+	} = paymentData || {};
+
+	const updateName = value => setPaymentData( { ...( paymentData || {} ), [ name ]: value } );
+	const updateAddress = value => setPaymentData( { ...( paymentData || {} ), [ address ]: value } );
+	const updateCity = value => setPaymentData( { ...( paymentData || {} ), [ city ]: value } );
+	const updateState = value => setPaymentData( { ...( paymentData || {} ), [ state ]: value } );
+	const updateProvince = value =>
+		setPaymentData( { ...( paymentData || {} ), [ province ]: value } );
+
+	return (
+		<React.Fragment>
+			<FormField
+				id={ location + '-name' }
+				type="Text"
+				label={ localize( 'Name' ) }
+				error={ eval( name + 'Error' ) }
```

Comment:
Using `eval()` is very dangerous and best avoided unless absolutely necessary. Better to type this out explicitly.